nimbus/src/enrollment.rs

```diff
@@ -436,81 +842,104 @@ mod tests {
         let db = Database::new(&tmp_dir)?;
         let mut writer = db.write()?;
         let nimbus_id = Uuid::new_v4();
+        let app_ctx = Default::default();
         let aru = Default::default();
-        let app_context = Default::default();
         assert_eq!(get_enrollments(&db)?.len(), 0);
         let exps = get_test_experiments();
-        insert_experiments(&db, &mut writer, exps)?;
 
         // User has opted out of new experiments.
-        // New experiments exist, but no enrolments have happened.
         set_global_user_participation(&db, &mut writer, false)?;
-        update_enrollments(&db, &mut writer, &nimbus_id, &aru, &app_context)?;
+
+        let merger = EnrollmentsMerger::new(&nimbus_id, &aru, &app_ctx);
+        let events = merger.evolve_enrollments_in_db(&db, &mut writer, &exps)?;
         writer.commit()?;
 
         let enrollments = get_enrollments(&db)?;
         assert_eq!(enrollments.len(), 0);
-        // We should see no experiment enrolments.
-        assert_eq!(get_experiment_enrollments(&db)?.len(), 0);
-        assert_eq!(
-            get_experiment_enrollments_with_status(&db, EnrollmentStatus::OptedOut)?.len(),
-            0
-        );
+        assert_eq!(events.len(), 0);
+        // We should see the experiment non-enrollments.
+        assert_eq!(get_experiment_enrollments(&db)?.len(), 2);
+        let disqualified_enrollments: Vec<ExperimentEnrollment> = get_experiment_enrollments(&db)?
```

Comment:
The name "disqualified enrollments" doesn't seem right to me here, you're filtering for `NotEnrolled` state rather than `Disqualified`.